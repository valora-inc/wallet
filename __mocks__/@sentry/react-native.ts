const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
  startTransaction: jest.fn(),
  nativeCrash: jest.fn(),
  wrap: jest.fn().mockImplementation((x) => x),
}

module.exports = Sentry
