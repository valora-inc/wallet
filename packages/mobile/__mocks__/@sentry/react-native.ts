const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
}

module.exports = Sentry
