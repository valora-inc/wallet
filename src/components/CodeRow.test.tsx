import { render } from '@testing-library/react-native'
import * as React from 'react'
import 'react-native'
import CodeRow, { CodeRowStatus } from 'src/components/CodeRow'

describe('CodeRow', () => {
  it('renders correctly for all CodeRowStatus states', () => {
    ;[
      CodeRowStatus.DISABLED,
      CodeRowStatus.INPUTTING,
      CodeRowStatus.PROCESSING,
      CodeRowStatus.RECEIVED,
      CodeRowStatus.ACCEPTED,
    ].map((status) => {
      const { toJSON } = render(
        <CodeRow
          status={status}
          inputValue={'test'}
          inputPlaceholder={'placeholder'}
          onInputChange={jest.fn()}
          shouldShowClipboard={jest.fn()}
        />
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
