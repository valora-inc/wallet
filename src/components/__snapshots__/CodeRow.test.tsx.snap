// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeRow renders correctly for all CodeRowStatus states 1`] = `
<View
  style={
    {
      "backgroundColor": "#F0F0F0",
      "borderColor": "#E6E6E6",
      "borderRadius": 3,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#505050",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 22,
      }
    }
  >
    placeholder
  </Text>
</View>
`;

exports[`CodeRow renders correctly for all CodeRowStatus states 2`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#FFF",
        "borderColor": "#E6E6E6",
        "borderRadius": 3,
        "borderWidth": 1,
        "flex": 0,
        "height": 50,
        "marginVertical": 5,
      },
    ]
  }
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    placeholder="placeholder"
    shouldShowClipboard={[MockFunction]}
    style={
      [
        {
          "color": "#2E3338",
          "flex": 1,
          "fontFamily": "Inter-Regular",
          "fontSize": 16,
          "lineHeight": 22,
          "paddingHorizontal": 0,
          "paddingVertical": 12,
        },
        undefined,
        undefined,
      ]
    }
    value="test"
  />
</View>
`;

exports[`CodeRow renders correctly for all CodeRowStatus states 3`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "borderColor": "#E6E6E6",
      "borderRadius": 3,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
      "position": "relative",
    }
  }
>
  <Text
    style={
      {
        "color": "#505050",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 22,
      }
    }
  >
    test...
  </Text>
  <ActivityIndicator
    color="#42D689"
    size="small"
    style={
      {
        "backgroundColor": "#FFF",
        "padding": 10,
        "position": "absolute",
        "right": 3,
        "top": 5,
      }
    }
  />
</View>
`;

exports[`CodeRow renders correctly for all CodeRowStatus states 4`] = `
<View
  style={
    {
      "backgroundColor": "#F8F9F9",
      "borderRadius": 3,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "#505050",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 22,
      }
    }
  >
    test...
  </Text>
</View>
`;

exports[`CodeRow renders correctly for all CodeRowStatus states 5`] = `
<View
  style={
    {
      "backgroundColor": "#F8F9F9",
      "borderRadius": 3,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "#505050",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 22,
      }
    }
  >
    test...
  </Text>
  <View
    style={
      {
        "backgroundColor": "#F8F9F9",
        "padding": 10,
        "position": "absolute",
        "right": 3,
        "top": 3,
      }
    }
  >
    <svg
      fill="none"
      height={20}
      style={{}}
      viewBox="0 0 24 24"
      width={20}
    >
      <path
        d="M20.5 7.33 9.186 18.643 4 13.458l1.33-1.33 3.856 3.847L19.17 6 20.5 7.33Z"
        fill="#42D689"
        style={{}}
      />
    </svg>
  </View>
</View>
`;
